test/plugin.test.js

```diff
@@ -0,0 +1,202 @@
+                'vendor', 'components']);
+            expect(chunkDependencies.qux).toEqual([
+                'components', 'feature-shared']);
+            expect(chunkDependencies.components).toEqual(['vendor']);
+            expect(chunkDependencies.vendor).toEqual([]);
+            expect(chunkDependencies['feature-shared']).toEqual([]);
+        });
+    });
+
+    it('should not generate empty chunks', () => {
+        const compiler = createCompiler({
+            foo: path.resolve(__dirname, 'fixtures/features/foo.js'),
+            baz: path.resolve(__dirname, 'fixtures/features/baz.js'),
+        });
+        compiler.outputFileSystem = new MemoryFileSystem();
+
+        return compile(compiler).then((stats) => {
+            const {chunks, modules} = stats.compilation;
+            const chunkToModulesMap = {};
+
+            for (const module of modules) {
+                for (const chunk of module.chunks) {
+                    if (!chunkToModulesMap.hasOwnProperty(chunk.name)) {
+                        chunkToModulesMap[chunk.name] = new Set();
+                    }
+                    chunkToModulesMap[chunk.name].add(module.portableId);
+                }
+            }
+
+            expect([...chunkToModulesMap.foo].sort()).toEqual([
+                'test/fixtures/features/foo.js',
+                'test/fixtures/features/percent.js',
+            ]);
+
+            expect([...chunkToModulesMap.baz].sort()).toEqual([
+                'test/fixtures/features/baz.js'
+            ]);
+
+            expect([...chunkToModulesMap.components].sort()).toEqual([
+                'test/fixtures/components/comp1.js',
+                'test/fixtures/components/comp2.js',
+                'test/fixtures/util/util.js',
+            ]);
+
+            expect([...chunkToModulesMap.vendor].sort()).toEqual([
+                'test/fixtures/vendor/dep1.js',
+                'test/fixtures/vendor/dep2.js',
+            ]);
+
+            expect(chunkToModulesMap['feature-shared']).toBeUndefined();
+
+            const chunkDependencies = {};
+            for (const chunk of chunks) {
+                const sortFn = (a, b) => {
+                    if (a.parents.includes(b)) {
+                        return 1;
+                    } if (b.parents.includes(a)) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                }
+                const orderedDeps = chunk.parents.sort(sortFn);
+                chunkDependencies[chunk.name] = orderedDeps.map(dep => dep.name);
+            }
+
+            expect(chunkDependencies.foo).toEqual(['vendor', 'components']);
+            expect(chunkDependencies.baz).toEqual(['vendor', 'components']);
+
+            expect(chunkDependencies.components).toEqual(['vendor']);
+            expect(chunkDependencies.vendor).toEqual([]);
+
+            expect(chunkDependencies['feature-shared']).toBeUndefined();
```

Comment:
`feature-shared` is empty b/c `feature/percent.js` only appears once across `foo` and `baz` entry points.